### ISSUE TITLE
sqitch deploy fails if database requires SSL=true

### ISSUE THREAD

**Issue by akabraham** [theory/sqitch#371]
Sqitch: 0.9996
Postgres: 9.6
Command: `sqitch deploy`
Error: ```FATAL:  password authentication failed for user "<USER>"
FATAL:  no pg_hba.conf entry for host "<HOST>", user "<USER>", database "<DATABASE>", SSL off```

It's unclear in the documentation how to specify SSL=required/true within the database URI.  Is this feature supported by sqitch?

**Comment by theory** [comment]
Ahhh TLS! I honestly don't know if there's a way to do it right now. I've yet to use TLS with Postgres (likely to change this year), so I'll look at making updates when I have to make it work, too. Unless you beat me to it.

In the meantime, I think you can put `?sslmode=require` in your URI and it *should* be used by the DBI  (see the [DBD::Pg connect docs](https://metacpan.org/pod/DBD::Pg#connect) for details. For the client, it looks like you can set the `PGSSLMODE` [environment variable](https://www.postgresql.org/docs/10/static/libpq-envars.html) to get it to do the right thing. Come to think of it, that might do the trick for the DBD::Pg connection too.

Let me know if any of that works!

**Comment by akabraham** [comment]
Thanks for the suggestions!  Both appending `?sslmode=require` to the database uri and setting `PGSSLMODE=require` will solve the `FATAL: no pg_hba.conf entry for ... database "<DATABASE>", SSL off` error.

Unfortunately, we are still getting a `FATAL: password authentication failed for user "<USER>"` error.  I've triple-checked that the username and password are indeed correct, so that shouldn't be the real problem.  This might now be a problem with our specific infrastructure setup and no longer an SSL/TLS issue.

For reference, our local test databases work perfectly with sqitch—it's only the VPN databases that are having issues.  The VPN databases are also behind PGBouncer, so that might complicate things further

**Comment by theory** [comment]
Are you able to connect via psql with the same credential?

I would not be surprised if deploys failed with PGBouncer. I think you are best off deploying directly to each Postgres service itself.

**Comment by akabraham** [comment]
Yes, connecting with the same credentials works with psql.  But thankfully I've figured it out.  I had `SQITCH_PASSWORD` set in my environment to one of our old db passwords, and apparently that overrides any password you put in the URI (either inline or in sqitch.conf).  Once that variable was unset, I was able to finally connect.

It would be good to make clear in the docs that `SQITCH_PASSWORD` takes precedence over anything you put in sqitch.conf.  Thanks again for your help!
